src/core/stdcpp/vector.d

```diff
@@ -633,6 +635,7 @@ extern(D):
                 _First = _Dest;
                 _Last = _Dest;
             }
+            static if (hasElaborateDestructor!T)
```

Comment:
C++ doesn't have static-if, `vector` always has a destructor.

Reply:
The rationale was that not having a destructor (even if it's empty) could prevent unnecessary stack-unwinding code from being inserted.

>in the event that the linker chooses one or the other, it may mix & match.

Since every function is extern(D) the mangling will never match. Wasn't that why you made them extern(D) when you wrote this?

Reply:
Oh yeah! That's exactly right... Haha, I haven't thought on this for like a year. I've forgotten all my working state >_<

So, I'm excited you're motivated to contribute to this. I think your work on the TODO's is cool, but like I say, I'm still not confident the test suite is comprehensive enough to deviate from the reference implementations. Are you interested in building out the test suite?

I think a good way to test this, would be to find OSS tests for the C++ library; maybe libc++ has the test suite source available, and adapt the tests to D.
The ultimate goal here, is to collapse the 3 distinct implementations into just one implementation.
The structs are different, and the allocation strategies are slightly different, so we'd keep versions of those functions, but the higher level functions, like the copy/move logic should be able to be centralised and not duplicated.

I wanted to do that last, since once you lose the reference code, it's very hard to refer back to the original C++ source and copy the functionality precisely. I'm not sure all of the internal functions are fleshed out yet, and the tests will help reveal what's missing.